netflix-roulette/src/pages/home/components/Header/components/SearchBar/SearchBar.test.tsx

```diff
@@ -0,0 +1,64 @@
+import * as React from "react";
+import { fireEvent, render, RenderResult } from "@testing-library/react";
+import "@testing-library/jest-dom";
+import { SearchBar } from "./SearchBar";
+import { ThemeProvider } from "styled-components";
+import { theme } from "@utils/theme";
+import { MemoryRouter } from "react-router-dom";
+
+const setSearchParamsMock = jest.spyOn(URLSearchParams.prototype, "get");
+
+const renderComponent = (children: React.ReactElement): RenderResult =>
+  render(
+    <ThemeProvider theme={theme}>
+      <MemoryRouter>{children}</MemoryRouter>
+    </ThemeProvider>
+  );
+
+describe("SearchBar", () => {
+  let component: RenderResult;
+
+  beforeEach(() => {
+    component = renderComponent(<SearchBar />);
+  });
+
+  test("SearchBar should be rendered correctly", () => {
+    const { asFragment } = component;
+
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  test("button test should be correct", () => {
+    const { getByText } = component;
+    const element = getByText("Search");
+
+    expect(element).toBeInTheDocument();
+  });
+
+  test("onClick event should be triggered", () => {
+    const { getByRole } = component;
+    fireEvent.click(getByRole("button"));
```

Comment:
For more precise simulation of user interactions, [it makes sense](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-using-testing-libraryuser-event) to use `userEvent()`.